pydoctor/sphinx.py

```diff
@@ -23,15 +32,22 @@ class SphinxInventory:
     Sphinx inventory handler.
     """
 
-    def __init__(self, logger, project_name=None):
+    def __init__(
+            self,
+            logger: Callable[..., None],
+            project_name: Optional[str] = None
+            ):
         """
         @param project_name: Dummy argument to stay compatible with
                              L{twisted.python._pydoctor}.
         """
-        self._links = {}
-        self.error = lambda where, message: logger(where, message, thresh=-1)
+        self._links: Dict[str, Tuple[str, str]] = {}
+        self._logger = logger
 
-    def update(self, cache, url):
+    def error(self, where: str, message: str) -> None:
```

Comment:
Why is this method added here?

(Ditto for the `info` method below)

Reply:
Because adding annotations to a lambda expression is not possible. The alternatives would be an inner function or a relatively complex `Callable` annotation, both of which are harder to read than a method, in my opinion.